sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -927,6 +928,121 @@ def sdc_pandas_dataframe_drop_impl(df, _func_name, args, columns):
     return sdc_pandas_dataframe_drop_impl(df, _func_name, args, columns)
 
 
+def df_getitem_str_literal_idx_main_codelines(self, idx):
+    """Generate main code lines for df.getitem"""
+    try:
+        col_idx = self.columns.index(idx)
+    except ValueError:
+        func_lines = ['  raise KeyError']
+    else:
+        col_data = f'data_{col_idx}'
+        func_lines = [
+            f'  {col_data} = get_dataframe_data(self, {col_idx})',
+            f'  return pandas.Series({col_data}, index=self._index, name=idx)'
```

Comment:
What if self._index is None? Is it ok for Series constructor?
Attribute index is supported for DataFrames, it handles 'no index' cases
```suggestion
            f'  return pandas.Series({col_data}, index=self.index, name=idx)'
```

Reply:
@akharche no. self._index is better

Reply:
It is ok to provide None as index to series

Reply:
Otherwise we would create index on each getitem. This is definitely something we don't want to happen